server/openslides/assignments/views.py

```diff
@@ -493,36 +493,28 @@ def create_votes_type_votes(self, data, poll, user):
                     continue
                 weight = Decimal(amount)
                 if config["users_activate_vote_weight"]:
-                    weight *= user.vote_weight
+                    weight *= vote_weight
                 vote = AssignmentVote.objects.create(
-                    option=option, user=user, weight=weight, value="Y"
+                    option=option, user=vote_user, weight=weight, value="Y"
                 )
                 inform_changed_data(vote, no_delete_on_restriction=True)
         else:  # global_no or global_abstain
             option = options[0]
-            weight = (
-                user.vote_weight if config["users_activate_vote_weight"] else Decimal(1)
-            )
+            weight = vote_weight if config["users_activate_vote_weight"] else Decimal(1)
             vote = AssignmentVote.objects.create(
-                option=option, user=user, weight=weight, value=data
+                option=option, user=vote_user, weight=weight, value=data
             )
             inform_changed_data(vote, no_delete_on_restriction=True)
             inform_changed_data(option)
             inform_changed_data(poll)
 
-        poll.voted.add(user)
```

Comment:
Where is this `add` now?

Reply:
This is done in the `assert_can_vote` method to have it handled the same way for all `vote` requests, see https://github.com/OpenSlides/OpenSlides/blob/04477d9ebd0488f35ecfbb02b8a548306a596251/server/openslides/poll/views.py#L237 (and ff.)